stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -625,7 +625,20 @@ extension ${Self}: BinaryFloatingPoint {
   /// almost never is.
   @_inlineable // FIXME(sil-serialize-all)
   public var ulp: ${Self} {
-    if !isFinite { return ${Self}.nan }
+%if bits == 32 or bits == 64:
+    guard _fastPath(isFinite) else { return .nan }
+    if _fastPath(isNormal) {
+      let bitPattern_ = bitPattern & ${Self}.infinity.bitPattern
+      return ${Self}(bitPattern: bitPattern_) / 0x1p${SignificandBitCount}
+    }
+#if arch(arm)
+    // On arm, flush positive subnormal values to 0.
+    return 0
+#else
+    return .leastNonzeroMagnitude
```

Comment:
Based on your points below, I'm going to delete the compilation condition and return `.leastNormalMagnitude * 0x1p-${SignificandBitCount}` here.